frontend/src/components/dashboard/dashboard.tsx

```diff
@@ -60,7 +88,15 @@ const Dashboard: FC = () => {
       {dataStatus === DataStatus.PENDING ? (
         <Spinner />
       ) : (
-        <CoursesList courses={courses} />
+        <CoursesList
+          courses={courses}
+          currentPage={
+            isPaginationClicked ? page : PaginationDefaultValue.DEFAULT_PAGE
+          }
```

Comment:
Why do we need this logic? As far as I know, `usePagination` should handle this

Reply:
In that case, we can have 2 `useEffect`s, one of them on page change, gets data for the search for that particular page, another one on search change, in that case we get data for that search term and first page :thinking: